CRM/Contribute/Form/AdditionalPayment.php

```diff
@@ -185,7 +185,7 @@ public function setDefaultValues() {
     }
 
     if ($this->_refund) {
-      $defaults['total_amount'] = abs($this->_refund);
+      $defaults['total_amount'] = number_format(abs($this->_refund), 2);
     }
```

Comment:
In this case the goal seems to be to have AT LEAST 2 decimal places , rather than 'not more than 2 decimal places. In this case it seems our primary goal is to pad it 

(aside: That function you have mentioned looks nasty. ie. the function has been hacked to be 'two functions in one' -
 it does something different if $onlyNumber is passed in which looks like a code smell to me)